src/app/common/h/sort-flights.ts

```diff
@@ -0,0 +1,91 @@
+import { Pipe, PipeTransform } from "@angular/core";
+import * as moment from 'moment';
+import { Flight } from '../entities/Flight';
+
+@Pipe({
+  name: "sortFlightsByDepartureAirport",
+})
+export class SortFlightsByDepartureAirport implements PipeTransform {
+  transform(items: Flight[], term, filterMetadata): any {
+    if (term === "All Airports") {
+      filterMetadata.count = items.length;
+      return items;
+    }
+
+    if (term) {
+      let filteredArray = items.filter(
+        (item) => item.departAirport.indexOf(term) !== -1
+      );
+      filterMetadata.count = filteredArray.length;
+      return filteredArray;
+    } else {
+      return items;
+    }
+  }
+}
+
+@Pipe({
+    name: "sortFlightsByArrivalAirport",
+  })
+export class SortFlightsByArrivalAirport implements PipeTransform {
+    transform(items: Flight[], term, filterMetadata): any {
+      if (term === "All Airports") {
+        filterMetadata.count = items.length;
+        return items;
+      }
+  
+      if (term) {
+        let filteredArray = items.filter(
+          (item) => item.arriveAirport.indexOf(term) !== -1
+        );
+        filterMetadata.count = filteredArray.length;
+        return filteredArray;
+      } else {
+        return items;
+      }
+    }
+  }
+
+@Pipe({
+name: "sortByFlightPrice",
+})
+export class SortByFlightPrice implements PipeTransform {
+    transform(items: Flight[], price, filterMetadata): any {
+        if (price === 100) {
```

Comment:
If the max price is changed in the component then this will need to be changed too.